### PR TITLE
Clear pending status on intakes on VBMS error

### PR DESCRIPTION
**Pull request by leikkisa**
Connects #5480

### CODE REVIEW

File: app/models/higher_level_review_intake.rb [hunk 1]
```diff
@@ -26,7 +26,12 @@ def complete!(request_params)
 
     detail.create_issues!(request_issues_data: request_params[:request_issues] || [])
 
-    detail.create_end_product_and_contentions!
+    begin
+      detail.create_end_product_and_contentions!
+    rescue StandardError
+      return clear_pending!
```
Comment (shanear) [inline app/models/higher_level_review_intake.rb]: I think we want to re raise the error here after we clear pending

File: app/models/higher_level_review_intake.rb [hunk 2]
```diff
@@ -26,7 +26,12 @@ def complete!(request_params)
 
     detail.create_issues!(request_issues_data: request_params[:request_issues] || [])
 
-    detail.create_end_product_and_contentions!
+    begin
```
Comment (shanear) [inline app/models/higher_level_review_intake.rb]: It also might be good to break this off into it's own private method

File: app/models/ramp_refiling_intake.rb [hunk 3]
```diff
@@ -27,12 +27,17 @@ def save_error!(code:)
   end
 
   def complete!(request_params)
-    return if complete? || pending?
+    return if complete_or_pending?
```
Comment (shanear) [inline app/models/ramp_refiling_intake.rb]: not sure if this is worth a method

File: app/models/higher_level_review_intake.rb (hunk 1 above)
Comment (leikkisa) [reply app/models/higher_level_review_intake.rb]: Okay, I think I understand (like re-raise the error).

File: app/models/ramp_refiling_intake.rb (hunk 3 above)
Comment (leikkisa) [reply app/models/ramp_refiling_intake.rb]: I had a cyclomatic complexity linting error :-/

Comment (leikkisa) [reply app/models/ramp_refiling_intake.rb]: Maybe breaking off the other piece into its own method will solve the Cyclomatic Complexity thing though!

File: app/models/higher_level_review_intake.rb (hunk 2 above)
Comment (shanear) [reply app/models/higher_level_review_intake.rb]: then we can use that method in `SupplementalClaim` and `HigherLevelReview`

**Pull request merged**